api_client/python/timesketch_api_client/aggregation.py

```diff
@@ -272,3 +276,172 @@ def generate_chart(self):
+                information.
+
+        Raises:
+            TypeError: if the dictionary does not contain the correct
+                information.
+        """
+        group_id = group_dict.get('id')
+        if not group_id:
+            raise TypeError('Group ID is missing.')
+        self.id = group_id
+        self.resource_uri = 'sketches/{0:d}/aggregation/group/{1:d}/'.format(
+            self._sketch.id, group_id)
+
+        self._name = group_dict.get('name', '')
+        self._description = group_dict.get('description', '')
+
+        self._how = group_dict.get('how')
+        if not self._how:
+            raise TypeError('How a group is connected needs to be defined.')
+
+        parameter_string = group_dict.get('parameters', '')
+        if parameter_string:
+            self._parameters = json.loads(parameter_string)
+
+        aggs = group_dict.get('agg_ids')
+        if not aggs:
+            raise TypeError('Group contains no aggregations')
+
+        aggs = json.loads(aggs)
+        if not isinstance(aggs, (list, tuple)):
+            raise TypeError('Aggregations need to be a list.')
+
+        self._aggregations = []
+        for agg_id in aggs:
+            agg_obj = Aggregation(self._sketch, self.api)
+            agg_obj.from_store(agg_id)
+            self._aggregations.append(agg_obj)
+
+    def from_store(self, group_id):
+        """Feed group data from a group ID.
+
+        Args:
+            group_id (int): the group ID to fetch from the store.
+
+        Raises:
+            TypeError: if the group ID does not exist.
+        """
+        self.id = group_id
+        resource_uri = 'sketches/{0:d}/aggregation/group/'.format(
+            self._sketch.id)
+        resource_data = self.api.fetch_resource_data(resource_uri)
+        for group_dict in resource_data.get('objects', []):
+            group_dict_id = group_dict.get('id')
+            if not group_dict_id:
+                continue
+            if group_dict_id == group_id:
+                self.from_dict(group_dict)
+                return
+        raise TypeError('Group ID not found.')
+
+    def generate_chart(self):
+        """Returns an altair Vega-lite chart."""
+        if not self._aggregations:
+            return altair.Chart()
+
+        data = self.lazyload_data()
+
+        meta = data.get('meta', {})
+        vega_spec = meta.get('vega_spec')
+
+        if not vega_spec:
+            return altair.Chart(pandas.DataFrame()).mark_point()
+
+        vega_spec_string = json.dumps(vega_spec)
+        return altair.Chart.from_json(vega_spec_string)
+
+    def get_charts(self):
+        """Returns a list of altair Chart objects from each aggregation."""
+        return [x.chart for x in self._aggregations]
+
+    def get_tables(self):
+        """Returns a list of pandas DataFrame from each aggregation."""
+        return [x.table for x in self._aggregations]
+
+    def to_pandas(self):
+        """Returns a pandas DataFrame."""
```

Comment:
Can you add some more words around what the DataFrame containes?

Reply:
ack...